flows/template_test.go

```diff
@@ -23,85 +26,85 @@ func TestTemplateTranslation(t *testing.T) {
+		tt := flows.NewTemplateTranslation(static.NewTemplateTranslation(channel, i18n.Locale("eng-US"), tc.Content, len(tc.Variables), "a6a8863e_7879_4487_ad24_5e2ea429027c"))
 		result := tt.Substitute(tc.Variables)
 		assert.Equal(t, tc.Expected, result, "%d: unexpected template substitution", i)
 	}
 }
 
-func TestTemplates(t *testing.T) {
-	channel1 := assets.NewChannelReference("0bce5fd3-c215-45a0-bcb8-2386eb194175", "Test Channel")
-	tt1 := static.NewTemplateTranslation(*channel1, i18n.Locale("eng"), "Hello {{1}}", 1, "")
-	tt2 := static.NewTemplateTranslation(*channel1, i18n.Locale("spa-EC"), "Que tal {{1}}", 1, "")
-	tt3 := static.NewTemplateTranslation(*channel1, i18n.Locale("spa-ES"), "Hola {{1}}", 1, "")
-	template := NewTemplate(static.NewTemplate("c520cbda-e118-440f-aaf6-c0485088384f", "greeting", []*static.TemplateTranslation{tt1, tt2, tt3}))
+func TestTemplate(t *testing.T) {
+	channel1 := test.NewChannel("WhatsApp 1", "+12345", []string{"whatsapp"}, []assets.ChannelRole{}, nil)
+	channel2 := test.NewChannel("WhatsApp 2", "+23456", []string{"whatsapp"}, []assets.ChannelRole{}, nil)
+	channel3 := test.NewChannel("WhatsApp 3", "+34567", []string{"whatsapp"}, []assets.ChannelRole{}, nil)
+	channel1Ref := assets.NewChannelReference(channel1.UUID(), channel1.Name())
+	channel2Ref := assets.NewChannelReference(channel2.UUID(), channel2.Name())
 
-	tas := NewTemplateAssets([]assets.Template{template})
+	tt1 := static.NewTemplateTranslation(channel1Ref, i18n.Locale("eng"), "Hello {{1}}", 1, "")
+	tt2 := static.NewTemplateTranslation(channel1Ref, i18n.Locale("spa-EC"), "Que tal {{1}}", 1, "")
+	tt3 := static.NewTemplateTranslation(channel1Ref, i18n.Locale("spa-ES"), "Hola {{1}}", 1, "")
+	tt4 := static.NewTemplateTranslation(channel2Ref, i18n.Locale("en"), "Hello {{1}}", 1, "")
+	template := flows.NewTemplate(static.NewTemplate("c520cbda-e118-440f-aaf6-c0485088384f", "greeting", []*static.TemplateTranslation{tt1, tt2, tt3, tt4}))
+
+	tas := flows.NewTemplateAssets([]assets.Template{template})
 
 	tcs := []struct {
-		UUID      assets.TemplateUUID
-		Channel   *assets.ChannelReference
-		Locales   []i18n.Locale
-		Variables []string
-		Expected  string
+		channel   *flows.Channel
+		locales   []i18n.Locale
+		variables []string
+		expected  string
 	}{
 		{
-			"c520cbda-e118-440f-aaf6-c0485088384f",
 			channel1,
 			[]i18n.Locale{"eng-US", "spa-CO"},
 			[]string{"Chef"},
 			"Hello Chef",
 		},
 		{
-			"c520cbda-e118-440f-aaf6-c0485088384f",
 			channel1,
 			[]i18n.Locale{"eng", "spa-CO"},
 			[]string{"Chef"},
 			"Hello Chef",
 		},
 		{
-			"c520cbda-e118-440f-aaf6-c0485088384f",
 			channel1,
 			[]i18n.Locale{"deu-DE", "spa-ES"},
 			[]string{"Chef"},
 			"Hola Chef",
 		},
 		{
-			"c520cbda-e118-440f-aaf6-c0485088384f",
-			nil,
-			[]i18n.Locale{"deu-DE", "spa-ES"},
+			channel1,
+			[]i18n.Locale{"deu-DE"},
 			[]string{"Chef"},
-			"",
+			"Hello Chef",
```

Comment:
this case is a little weird because it's just defaulting to the first translation... but it does seem better to generally return something here than creating a message without a template that probably going to fail.